From cbbe2ef426a903fc2ae9dc784e61d0208fe52a87 Mon Sep 17 00:00:00 2001
From: Shivani Agarwal <shivania2@vmware.com>
Date: Fri, 9 Sep 2022 14:16:15 +0000
Subject: [PATCH 24/29] Fix kubernetes denials for K8's deployment with
 containerd

Changes are added to resolve following kubernetes  denials

avc:  denied  { read } for  pid=6084 comm="kube-controller"
path="/etc/kubernetes/pki/front-proxy-ca.crt" dev="sda3" ino=1049304
scontext=system_u:system_r:container_t:s0:c221,c620
tcontext=system_u:object_r:kubernetes_file_t:s0 tclass=file permissive=0
---
 policy/modules/contrib/container.if | 43 +++++++++++++++++++++++++++++
 policy/modules/contrib/container.te |  2 ++
 2 files changed, 45 insertions(+)

diff --git a/policy/modules/contrib/container.if b/policy/modules/contrib/container.if
index a77e673..adcd6da 100644
--- a/policy/modules/contrib/container.if
+++ b/policy/modules/contrib/container.if
@@ -185,6 +185,49 @@ interface(`container_runtime_read_tmpfs_files',`
 	read_lnk_files_pattern($1, container_runtime_tmpfs_t, container_runtime_tmpfs_t)
 ')
 
+########################################
+## <summary>
+##      Read container runtime kubernetes files.
+## </summary>
+## <param name="domain">
+##      <summary>
+##      Domain allowed access.
+##      </summary>
+## </param>
+#
+interface(`container_runtime_read_kubernetes_files',`
+        gen_require(`
+                type kubernetes_file_t;
+        ')
+
+        files_search_var_lib($1)
+        read_files_pattern($1, kubernetes_file_t, kubernetes_file_t)
+')
+
+########################################
+## <summary>
+##      Allow process to watch bin files.
+## </summary>
+## <desc>
+##      <p>
+##      Allow the specified domain to watch kubernetes files
+##      </p>
+## </desc>
+## <param name="domain">
+##      <summary>
+##      Domain allowed access.
+##      </summary>
+## </param>
+#
+interface(`container_runtime_watch_kubernetes_files',`
+        gen_require(`
+                type kubernetes_file_t;
+        ')
+
+         allow $1 kubernetes_file_t:dir watch_dir_perms;
+         allow $1 kubernetes_file_t:file watch_file_perms;
+')
+
 ########################################
 ## <summary>
 ##	Manage container share files.
diff --git a/policy/modules/contrib/container.te b/policy/modules/contrib/container.te
index 3c23c4d..95a888a 100644
--- a/policy/modules/contrib/container.te
+++ b/policy/modules/contrib/container.te
@@ -1299,3 +1299,5 @@ kernel_mounton_core_if(container_engine_t)
 kernel_mounton_proc(container_engine_t)
 kernel_mounton_systemd_ProtectKernelTunables(container_engine_t)
 term_mount_pty_fs(container_engine_t)
+container_runtime_read_kubernetes_files(container_t)
+container_runtime_watch_kubernetes_files(container_t)
-- 
2.45.1

